acstools/acsphotcte.py

```diff
@@ -0,0 +1,282 @@
+            headers=self._api_credentials
+        )
+
+        status_code = response.status_code
+        content = json.loads(response.content)
+        if status_code > 400:
+            content = (f"Status Code: {status_code}\n"
+                       f"Message: {content['message']}")
+        return content
+
+    def _check_inputs(self, **inputs):
+        """Check the data types of the user supplied inputs
+
+        JSON is very finicky and the native python json module only works
+        with native python types (int, float, list, str). Hence, before
+        we serialize the inputs as a JSON object we need to compare all the
+        input data types to their python counterparts.
+
+        Parameters
+        ----------
+        inputs : dict
+            Dictionary containing all of the user supplied inputs
+
+        Returns
+        -------
+        inputs : dict, str, `None`
+            If the inputs conform to what is expected, it returns the original
+            dictionary with all of the inputs converted to their python
+            counterparts for JSON serialization. If one of the inputs fails,
+            it returns the corresponding key. If the iterable inputs are not
+            all the same length, it returns `None`.
+        """
+        iterable_lengths = []
+        for key in inputs.keys():
+            # Check if the input is Iterable and not a list  (e.g. np.array)
+            # We also need to make sure the input is not a str, since str are
+            # Iterable in Python.
+            if isinstance(inputs[key], Iterable) and \
+                    not isinstance(inputs[key], (str, list)):
+                # Convert the array to a python list
+                inputs[key] = list(inputs[key])
+
+                # Check if the first element of the input array is a float
+                if not isinstance(inputs[key][0], float):
+                    LOG.info('Converting elements..')
+                    # Convert the array of elements to floats
+                    inputs[key] = list(map(float, inputs[key]))
+                # Save the length for comparision later
+                iterable_lengths.append(len(inputs[key]))
+            # Check if the input is a python float, not an np.float32
+            # Handles the case of str inputs as well
+            elif not isinstance(inputs[key], float):
+                try:
+                    inputs[key] = float(inputs[key])
+                except TypeError as e:
+                    # TypeError during the conversion, print the info and
+                    # return the key it failed on
+                    LOG.error(e)
+                    return key
+                else:
+                    if key == 'radius' and inputs[key] not in [3., 5.]:
+                        LOG.error('Submitted radii not supported...')
+                        return key
+        # the last step is to ensure all the iterable inputs are the same length
+        nl = "\n"
```

Comment:
If you ever plan to support Windows...
```suggestion
        nl = os.linesep
```